jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt

```diff
@@ -139,118 +113,117 @@ class DotNetSamDebugSupport : SamDebugSupport {
     ): Promise<XDebugProcessStarter?> {
         val frontendPort = debugPorts[0]
         val backendPort = debugPorts[1]
-
         val promise = AsyncPromise<XDebugProcessStarter?>()
-        val project = environment.project
 
         // Define a debugger lifetime to be able to dispose the debugger process and all nested component on termination
-        val debuggerLifetimeDefinition = project.defineNestedLifetime()
+        val debuggerLifetimeDefinition = environment.project.defineNestedLifetime()
         val debuggerLifetime = debuggerLifetimeDefinition.lifetime
 
         val scheduler = RdDispatcher(debuggerLifetime)
-        val startInfo = createNetCoreStartInfo(state)
 
         val debugHostAddress = InetAddress.getByName(debugHost)
 
         val protocol = Protocol(
+            name = environment.runProfile.name,
             serializers = Serializers(),
             identity = Identities(IdKind.Client),
             scheduler = scheduler,
             wire = SocketWire.Client(debuggerLifetime, scheduler, hostAddress = debugHostAddress, port = frontendPort, optId = "FrontendToDebugWorker"),
             lifetime = debuggerLifetime
         )
 
+        val workerModel = RiderDebuggerWorkerModelManager.createDebuggerModel(debuggerLifetime, protocol)
+
         val executionResult = state.execute(environment.executor, environment.runner)
+        val console = TextConsoleBuilderFactory.getInstance().createBuilder(environment.project).console
+        val samProcessHandle = executionResult.processHandler
+        console.attachToProcess(samProcessHandle)
+
+        // If we have not started the process's notification system, start it now.
+        // This is needed to pipe the SAM output to the Console view of the debugger panel
+        if (!samProcessHandle.isStartNotified) {
```

Comment:
what's the case where the process has already started?

Reply:
Doesn't hurt. It is already pumping events out to the process listeners.

Preventing the call prevents a LOG.error() that gets invoked if the AtomicReference is already set to started state.